src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -435,9 +442,11 @@ std::vector<std::optional<PosHashTable<HashedType>>> build(const RadixContainer<
       }
     };
 
-    if (radix_bits == 0) {
+    if (radix_bits == 0 || JoinHash::JOB_SPAWN_THRESHOLD > elements_count) {
       // Without radix partitioning, only a single hash table will be written. Parallelizing this would require a
       // concurrent hash table, which is likely more expensive.
+      // If the size (number of elements) of the partition lies under the threshold, the execution in parallel is
```

Comment:
I still don't like this "lies under". To me is doesn't sound correct. Would use "is below".

Reply:
I'd change the comment to
```
Execute the insertion in the hash table sequentially when we do not radix partition (i.e., 0 radix bits) or the number of elements is too small.
Without radix partitioning, only a single hash table will be written. Parallelizing this would require a
concurrent hash table, which is likely more expensive.
```